data/collector.go

```diff
@@ -311,7 +327,7 @@ func (c *collector) writeCaptureResults() error {
 func (c *collector) logCaptureErrs() {
 	for err := range c.captureErrors {
 		now := c.clock.Now().Unix()
-		if c.closed {
+		if c.closeStarted.Load() {
```

Comment:
IIUC, the check on `c.closed` in the old code would always see `false`. Ironically, if this ever saw "true" in the intended way to avoid the log line -- Go would have detected this to be a data race.